opm/input/eclipse/Schedule/Well/WCYCLE.hpp

```diff
@@ -0,0 +1,146 @@
+/*
+  Copyright 2023 Equinor.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef OPM_WCYCLE_HPP
+#define OPM_WCYCLE_HPP
+
+#include <functional>
+#include <map>
+#include <unordered_map>
+#include <string>
+#include <vector>
+
+namespace Opm {
+
+class DeckRecord;
+class WellMatcher;
+
+class WCYCLE
+{
+public:
+    /// A vector of wells and efficiency factor scalings to apply.
+    using WellEfficiencyVec = std::vector<std::pair<std::string, double>>;
+    /// A map from well name to a time stamp.
+    using WellTimeMap = std::map<std::string, double>;
+    /// A vector of wells and their open status.
+    using WellStatusMap = std::map<std::string, bool>;
```

Comment:
I didn't expect a `bool` when seeing the word `Status` here.  Maybe `WellIsOpenMap` or something of the kind could be a bit more explanatory.

Reply:
sure. it's been under a few revisions so there was more than a bool at some point. but i agree it's easily confused with other uses of WellStatus so I have renamed it.